headers.go

```diff
@@ -440,6 +423,38 @@ func validateHeaderLabel(h map[interface{}]interface{}) error {
 	return nil
 }
 
+// normalizeLabel tries to cast label into a int64 or a string.
+// Returns (nil, false) if the label type is not valid.
+func normalizeLabel(label interface{}) (interface{}, bool) {
+	switch v := label.(type) {
+	case int:
+		label = int64(v)
+	case int8:
+		label = int64(v)
+	case int16:
+		label = int64(v)
+	case int32:
+		label = int64(v)
+	case int64:
+		label = int64(v)
+	case uint:
+		label = int64(v)
+	case uint8:
+		label = int64(v)
+	case uint16:
+		label = int64(v)
+	case uint32:
+		label = int64(v)
+	case uint64:
+		label = int64(v)
```

Comment:
should we intercept the potential overflow here?

Reply:
I'll rather do it in another PR. This is clearly an issue but requires its own discussion.

Reply:
Sure. #79 for tracking this.